### DISCUSSION TITLE
rg: unrecognized file type

### DISCUSSION THREAD

**Discussion by gengdz** [ibhagwan/fzf-lua#1333] [Q&A]
It works fine when I use it directly in zsh
```bash
rg 'color' -g src/test/**/* -ttsx
```

but in fzf-lua,
```bash
color --  -g 'src/test/**/*' -ttsx
```

 it throws an error: `rg: unrecognized file type: tsx`

And I noticed that the `-t` option doesn't seem to work. Even when I wrote `-tts`, it didn't work, although it didn't throw an error, other file types still appeared.

---

`~/.config/.ripgreprc` config:

```bash
--type-add
tsx:*.tsx

```

my fzf-lua config: [my nvim config](https://github.com/gengdz/dotfiles/blob/053bf506b123dfcbb05e2c3f076f30d8927b0496/nvim/lua/plugins/editor.lua#L70)

```lua
{
    "ibhagwan/fzf-lua",
    event = "VeryLazy",
    opts = function(_, opts)
      local config = require("fzf-lua.config")
      local actions = require("fzf-lua.actions")

      opts.grep = vim.tbl_deep_extend("force", opts.grep, {
        no_header = true,
        rg_glob = true,
        rg_glob_fn = function(query, _opts)
          local regex, flags = query:match("^(.-)%s%-%-(.*)$")
          -- If no separator is detected will return the original query
          return (regex or query), flags
        end,
        fzf_opts = {
          ["--history"] = vim.fn.stdpath("data") .. "/fzf-lua-grep-history",
        },
      })

      opts.files = vim.tbl_deep_extend("force", opts.files, {
        no_header = true,
        fzf_opts = {
          ["--history"] = vim.fn.stdpath("data") .. "/fzf-lua-files-history",
        },
      })

      config.defaults.keymap.fzf["ctrl-j"] = "next-history"
      config.defaults.keymap.fzf["ctrl-k"] = "prev-history"
      config.defaults.keymap.fzf["ctrl-n"] = "down"
      config.defaults.keymap.fzf["ctrl-p"] = "up"
      config.defaults.actions.files["ctrl-t"] = actions.file_tabedit
    end,
  },
```

**Comment by ibhagwan** [comment]
That’s because the default parser only supports globs, it translates each argument to `-g` (use `debug=true` and you’ll see the generated command)`, to filter  test files with the defaults you can only send:
```
color --  src/test/**/*
```

If you wish to send raw arguments this way customize your `rg_glob_fn`:
https://github.com/ibhagwan/fzf-lua/wiki#live-grep-custom

**Reply by gengdz** [reply]
I checked the documentation, and my configuration is written according to the instructions in the documentation.

rg_glob_fn 
```lua
 rg_glob_fn = function(query, _opts)
          local regex, flags = query:match("^(.-)%s%-%-(.*)$")
          -- If no separator is detected will return the original query
          return (regex or query), flags
        end,
```
 is it that my `rg_glob_fn` is written incorrectly?

**Reply by ibhagwan** [reply]
It should be under `grep`, also add `debug=true` so you can see the underlying command and know if you did this right.
